### ISSUE TITLE
typeFilter not working as expected

### ISSUE THREAD

**Issue by manemarron** [SteveGilham/altcover#19]
I am trying to instrument a net core 2.0 project.  I am running version 3.0.475 on a Windows host. This is the command I am running:
```
$ dotnet path\to\AltCover.dll -i=.\bin\Debug\netcoreapp2.0 -t=Program;Startup --save --opencover
```

Every time I execute the command, the next error appears:
```
    ERROR *** Instrumentation phase failed
    Specified argument was out of the range of valid values.
    Parameter name: target
    Details written to path\to\__Saved\AltCover-2018-05-21--11-52-12.log
```

These are the contents of the log file:
```
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: target
   at Mono.Cecil.Cil.ILProcessor.InsertBefore(Instruction target, Instruction instruction)
   at AltCover.Instrument.InsertVisit(Instruction instruction, ILProcessor methodWorker, MethodReference recordingMethodRef, String moduleId, Int32 point) in C:\projects\altcover\AltCover\Instrument.fs:line 335
   at AltCover.Instrument.VisitBranchPoint$cont@431(Context state, GoTo branch, Unit unitVar) in C:\projects\altcover\AltCover\Instrument.fs:line 453
   at AltCover.Instrument.VisitBranchPoint(Context state, GoTo branch) in C:\projects\altcover\AltCover\Instrument.fs:line 486
   at AltCover.Instrument.InstrumentationVisitorWrapper(FSharpFunc`2 core, Context state, Node node) in C:\projects\altcover\AltCover\Instrument.fs:line 611
   at AltCover.Visitor.stateful@564-1.Invoke(T node) in C:\projects\altcover\AltCover\Visitor.fs:line 567
   at Microsoft.FSharp.Primitives.Basics.List.mapToFreshConsTail[a,b](FSharpList`1 cons, FSharpFunc`2 f, FSharpList`1 x)
   at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult](FSharpFunc`2 mapping, FSharpList`1 x)
   at Microsoft.FSharp.Collections.SeqModule.Fold[T,TState](FSharpFunc`2 folder, TState state, IEnumerable`1 source)
   at AltCover.Visitor.Visit(FSharpList`1 visitors, IEnumerable`1 assemblies) in C:\projects\altcover\AltCover\Visitor.fs:line 554
   at AltCover.Main.result@330-4.Invoke(Unit unitVar0) in C:\projects\altcover\AltCover\AltCover.fs:line 344
   at AltCover.CommandLine.doPathOperation[a](FSharpFunc`2 f, a defaultValue, Boolean store) in C:\projects\altcover\AltCover\CommandLine.fs:line 127
ActualValue =
<null>
ParamName =
"target"
Data =
seq []
InnerException =
<null>
TargetSite =
Void InsertBefore(Mono.Cecil.Cil.Instruction, Mono.Cecil.Cil.Instruction)
HelpLink =
<null>
Source =
"Mono.Cecil"
HResult =
-2146233086
```

~~**Note**: When I remove the semicolon and just add one type, the command works fine~~
___
**Edit**: The `Startup` type is the one generating the error. When I only add `-t=Startup`, the command fails with the same error.

**Comment by SteveGilham** [comment]
Just to clarify -- when you say "just add one type", you can do either `-t=Program` or `-t=Startup`, and each of those work, but not when you specify both at once?

**Comment by manemarron** [comment]
Exactly

**Comment by manemarron** [comment]
@SteveGilham I hadn't tested only with Startup and that is the one that generates the error. I don't know why that may be.

**Comment by SteveGilham** [comment]
I think I have a repro and a diagnosis -- filtering is not being correctly applied to the branch points in a method, which is why the problem appears where it does.

**Comment by SteveGilham** [comment]
Probable fix in build [pre-release build 479](https://ci.appveyor.com/project/SteveGilham/altcover/build/artifacts), if you want to give it a shot.

**Comment by manemarron** [comment]
Will do. I'll let you know the outcome.

**Comment by manemarron** [comment]
It seems to work fine!

**Comment by SteveGilham** [comment]
Super.  The fix will be in a formal release in the next few days.

**Comment by SteveGilham** [comment]
Fix should be in the new 3.0.488 release
